### PR TITLE
PS3 Build fix

### PR DESCRIPTION
**Pull request by crystalct**

### CODE REVIEW

**Comment by realnc** [comment]
I'm going to make a couple comments for minor changes shortly, but before that here's how to properly commit the changes in git:

Instead of creating new commit, you should just amend the current commit. For example:

    git add  libretro/libretro_dosbox.h
    git commit --amend

Since you are changing an existing commit, you need to force push to your repo:

    git push --force

GitHub will recognize the forced push and will update the PR automatically.

If you are in a situation like in the previous PR where you had several commits and you want to squash them, you would do:

    git rebase -i <commit just prior to your first one>

then mark the subsequent commits after your first one with `f` (meaning "fixup"). This will squash them and then you could have forced pushed.

File: libretro/libretro_dosbox.h [hunk 1]
```diff
@@ -9,6 +9,32 @@
...
+#define S_ISDIR(mode)  (((mode) & S_IFMT) == S_IFDIR)
+#define getenv(a)	"/dev_hdd0/SSNE10000/USRDIR/cores/"
+#define	getcwd(a,b)	"/dev_hdd0/SSNE10000/USRDIR/"
+struct timeb {
+    time_t          time;
+    unsigned short  millitm;
+    short           timezone;
+    short           dstflag;
+};
+char *dirname(char *path);
```
Comment (realnc) [inline libretro/libretro_dosbox.h:33]: I believe this isn't needed anymore.

File: libretro/libretro_dosbox.h [hunk 2]
```diff
@@ -9,6 +9,32 @@
...
+int gettimeofday(struct timeval* tv, void* /*tz*/);
```
Comment (realnc) [inline libretro/libretro_dosbox.h:34]: Should be: `int gettimeofday(timeval* tv, void* /*tz*/);` unless your compiler complaints? (You don't need `struct` in C++ unless there's a name conflict.)

File: libretro/libretro_dosbox.h [hunk 3]
```diff
@@ -9,6 +9,32 @@
...
+int access(char *fpath, int /*mode*/);
```
Comment (realnc) [inline libretro/libretro_dosbox.h:35]: Should be: `int access(const char *fpath, int /*mode*/);`

**Review by realnc — CHANGES_REQUESTED** [review]

**Comment by crystalct** [comment]
All done.
Thanks for the patience :P

**Comment by crystalct** [comment]
Last minute modification:
added `CXXFLAGS += -D__STDC_LIMIT_MACROS` so g++ Know macros UINTPTR_MAX and SIZE_MAX

File: libretro/libretro.cpp [hunk 4]
```diff
@@ -40,6 +39,28 @@
...
+
+    tv->tv_sec = time / 1000000;
+    tv->tv_usec = time - (tv->tv_sec * 1000000);  // implicit rounding will take care of this for us
+    return 0;
+}
+int access(const char *fpath, int /*mode*/)
+{
+    struct stat buffer;   
+    return stat(fpath, &buffer); 
+}
```
Comment (realnc) [inline libretro/libretro.cpp:57]: Last nitpick:

Signature also be `int gettimeofday(timeval* tv, void* /*tz*/)`

 Also, it's best to move these function implementations at the end of the file (after the Vita and PSP functions.) So simply protect the `<libgen.h>` include with `#ifndef __CELLOS_LV2__` and leave it where it originally was, and move the functions to the end of the file protected by `#ifdef __CELLOS_LV2__`.

**Review by realnc — CHANGES_REQUESTED** [review]

**Comment by realnc** [comment]
Other than the last nitpick, looks good. I'll merge it afterwards. :-)

Btw, are these fixes intended for PSL1GHT?

**Comment by crystalct** [comment]
Done.
About PSl1GHT, untill psl1ght Retroarch layer doens't work, cores can work only compiled with old Sony SDK. When PSL1GHT version of Retroarch will be accetable, we can adjust PSL1GHT compilations with pleasure.

**Comment by realnc** [comment]
Thanks! Merged. You probably intend to do a PR on libretro-super to enable the build on the buildbot?

**Comment by crystalct** [comment]
Since long time, i'm asking to retroarch guys to reopen PS3 compilations and downloads....
PS3 version is hosted by xbins.orgs and they stopped since August 2020.

**Pull request merged**